### PR TITLE
Filter Errors

### PR DESCRIPTION
**Pull request by gregfromstl**
Changes:
- Do not display errors when on the wrong network
- Add flag to disable throwing errors when a contract call reverts
- Update possibly-reverting contract calls to not throw errors when they do and use the default value instead

### CODE REVIEW

File: ui/lib/use-handle-error.js [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 
-export function useHandleError([{ data, error, loading }, method]) {
+export function useHandleError(
+  [{ data, error, loading }, method],
+  throwOnRevert = true // for some contract interactions, a reverted call is not an error
+) {
   const errorContext = useErrorContext()
   useEffect(() => {
-    errorContext.addError([error])
+    if (throwOnRevert) errorContext.addError([error])
```
Comment (aahna-ashina) [inline ui/lib/use-handle-error.js:10]: Suggestion for better readability: Use a separate line for the code inside the if statement:

```javascript
if (throwOnRevert) {
  errorContext.addError([error])
}
```

**Review by aahna-ashina — APPROVED** [review]

**Comment by luisivan** [comment]
Looks good to me! Just one more thing, we migrated to wagmi3 on the `develop` branch, would be worth it to make the couple changes needed for it to work with wagmi3 (I think mainly changing the use of `useNetwork`).

**Comment by gregfromstl** [comment]
@luisivan merged with develop, resolved conflicts, and upgraded the useNetwork hooks

**Comment by luisivan** [comment]
Awesome! LGTM except that the switch network banner is still appearing when on mainnet (testing on Vercel deployment https://app-git-fork-gregfromstl-greg-dw-34filter-the-er-d97935-nation3.vercel.app)

**Comment by luisivan** [comment]
Also claimable liquidity rewards are showing as zero in that deployment, weird

**Comment by luisivan** [comment]
Cool, just a couple more bugs:
- When the user rejects transactions:
Screen Shot 2022-05-17 at 8 27 12 PM
- Still cannot see claimable rewards

Apart from that, it seems that everything else is working!

**Pull request merged**